### PR TITLE
Added cancel Request and updated queries

### PR DESCRIPTION
**Pull request by AseelaAbdulSattar**

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -20,10 +19,10 @@ def create
     redirect_to new_friendship_path
   end
 
-  def destroy
+  def cancel_request
     redirect_to requests_sent_friendships_path
   end
-
+
```
Comment (hassanakram) [inline app/controllers/friendships_controller.rb]: Add preference to remove trailing spaces on save in your text editor i.e VSCode

File: app/models/user.rb [hunk 2]
```diff
@@ -10,14 +10,14 @@ def self.all_except(user)
   end
   
   def my_friends
-    User.where(id: self.friendships.true_friends.pluck(:friend_id))
+    User.where(id: friendships.true_friends.pluck(:friend_id))
```
Comment (hassanakram) [inline app/models/user.rb]: why can't we use `friends` relationship?

Comment (hassanakram) [reply app/models/user.rb]: https://stackoverflow.com/questions/20307874/what-is-the-equivalent-of-the-has-many-conditions-option-in-rails-4

File: app/views/friendships/friend_requests.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<% if @friends_req.empty?%>
+  <h2>No Friend Requests</h2>
+<% else @friends_req %>
+  <h1> Friend Requests </h1>
+  <%= render partial: "friend", collection: @friends_req, :locals => {:greeting => '1'}%>
```
Comment (hassanakram) [inline app/views/friendships/friend_requests.html.erb]: Please use new syntax & always a space after `{` & before `}` i.e 
```
locals: { greeting: '1' }
```

For what purpose are we using this variable `greeting`?

Comment (AseelaAbdulSattar) [reply app/views/friendships/friend_requests.html.erb]: 3 views were using same partial, so to differentiate between them I used this variable

File: app/views/friendships/index.html.erb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <% if @friends.empty? %>
   <h2>No Friends Exists</h2>
-<% elsif !@friends.empty?%>
+<% else @friends %>
   <h1>My Friends</h1>
-  <%= render partial: "friend", collection: @friends, :locals => {:greeting => '0'}%>
+  <%= render partial: "friend", collection: @friends, locals: { is_friend: '1' }%>
```
[the remark this answers is not in the document]
Comment (hussainakram) [reply app/views/friendships/index.html.erb]: @AseelaAbdulSattar   `is_friend` should be boolean here, as mentioned above

File: app/views/friendships/_friend.html.erb [hunk 5]
```diff
@@ -1,21 +1,16 @@
...
   <tr>
     <td><%= friend.name%></td>
-    <td><hr/></td>
     <td><%= friend.email%></td>
-    <td><hr/></td>
     <td><%= friend.id%></td>
-    <td><hr/></td>
-    <td><%= link_to 'Unfriend', unfriend_friendship_path(friend), method: :unfriend, data: { confirm: 'Are you sure?' }  if is_friend == '1' %></td>
-    <td><%= link_to 'Cancel Request', cancel_request_friendship_path(friend), method: :cancel_request, data: { confirm: 'Are you sure?' }  if is_friend == '0' %></td>
+    <td><%= link_to 'Unfriend', unfriend_friendship_path(friend), method: :post, data: { confirm: 'Are you sure?' }  if is_friend == true %></td>
```
Comment (hussainakram) [inline app/views/friendships/_friend.html.erb]: you don't have to do `is_friend == true` for a boolean variable. Rails is intelligent enough to detect that.
so you can replace it with just `if is_friend`

Please change it in other places too

**Pull request merged**